WildShape-Sheets-API/Services/AuthService.cs

```diff
@@ -29,19 +30,25 @@ public AuthService(IOptions<WildshapeSheetsDBSettings> wildshapeSheetsDBSettings
 
         public  string? Authenticate(string email, string password)
         {
-            var user = users.Find(user => user.Email == email && user.Password == password).FirstOrDefault();
 
-            if (user == null)
-            {
-                return null;
+            var user = users.Find(user => user.Email == email).FirstOrDefault();
+
+            if (user != null && user.Password != null && user.Salt != null) {
+                if (userService.VerifyPassword(password, user.Password, user.Salt)) {
+                    // Password is valid, proceed with successful login
+                    Console.WriteLine("Login valid");
+                } else {
+                    Console.WriteLine("Login invalid");
+                    return null;
+                }
             }
 
             const string secretKey = "Shared secret key that no one ever knew";
```

Comment:
Don't want secretKey stored in the codebase. For this we want to use a combination of appsettings.json and environment variables.

I'm not sure of the impementation details, but if it is similar to how other systems work, the appsettings.json is not checked in with the codebase, and is only used to grab the environment variable values for when you run the app locally. The server environment you are eventually deploying to will have a mechanism for defining the environment variables in that specific environment. Then in your code, you do something like:
string valueNotToStore = Environment.GetEnvironmentVariable("VALUE_NOT_TO_STORE");

I'm sure about the details on how to set up appsetting.json, but I see you have one, so maybe you know how it works. Pretty much everything in that file you don't want exposed, so best to add appsetting.json to your .ignore.